code/game/objects/structures/headpike.dm

```diff
@@ -40,6 +40,9 @@
 	return ..()
 
 /obj/structure/headpike/update_name()
+	if(!victim)
+		name = initial(name)
+		return ..()
 	name = "[victim.real_name] on a [spear]"
```

Comment:
name = victim ? "[victim.real_name] on a [spear]" : initial(name)

Reply:
idk, if you're sure. I'm not a huge fan of ternaries but maybe it's fine here